src/elisp/treemacs-mouse-interface.el

```diff
@@ -48,6 +52,17 @@ Must be bound to a mouse click, or EVENT will not be supplied."
        :no-error            t))
     (treemacs--evade-image)))
 
+(defun treemacs-downleftclick-action (event)
+  "Select the window and record the node's path."
+  (interactive "e")
+  (when (eq 'down-mouse-1 (elt event 0))
+    (run-hooks 'mouse-leave-buffer-hook)
+    (select-window (elt (elt event 1) 0))
+    (goto-char (posn-point (cadr event)))
+    (setq treemacs-last-selected-button (treemacs-safe-button-get (treemacs-current-button) :path))
```

Comment:
The -safe-button-get is only needed when current-buffer and the button's buffer are different.

Reply:
So, what should be used instead?

Reply:
Nothing, just the the -current-button as is.